dts/bindings/ipc/nordic,nrf-ipct-common.yaml

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: Nordic IPCT (Interprocessor Communication Transceiver)
+
+include: base.yaml
+
+properties:
+  reg:
+    required: true
+
+  channels:
+    type: int
+    required: true
+    description: Number of channels implemented by the IPCT instance.
+
+  source-channel-links:
+    type: array
+    description: |
+      Mapping of IPCT channels that are mapped between two IPCT instances on
+      separate domains, in which a channel on this IPCT node is considered
+      the source. This array is then comprised of a 3-integer-wide "unit"
+      that defines one connection of the mapping. The format of this unit
+      is <source_channel sink_domain_id sink_channel>. Units are sequential
+      in the array, therefore requiring the length of this property to be
+      a factor of 3.
+
+      For example, if channel 2 is to be mapped to Secure Domain (ID: 1) IPCT
+      channel 4, then the array "unit" would be <2 1 4>.
```

Comment:
note: I'd provide definitions for domain identifiers, otherwise reading dt files becomes a bit cumbersome, specially if future SoCs use other ids. These could be defined in the main/root DT file itself (no need to include anything)

Reply:
Added. Please take a look.